### PR TITLE
build: include share/rpcauth in release tarball

### LINKED ISSUE #19081: Build/Docs: -help text references rpcauth script not present in binary release
The help text says "a canonical python script is included in `share/rpcauth`":

    $ bitcoind -help | grep -A3 rpcauth
      -rpcauth=<userpw>
           Username and HMAC-SHA-256 hashed password for JSON-RPC connections. The
           field <userpw> comes in the format: <USERNAME>:<SALT>$<HASH>. A
           canonical python script is included in share/rpcauth. The client
           then connects normally using the
           rpcuser=<USERNAME>/rpcpassword=<PASSWORD> pair of arguments. This
           option can be specified multiple times

This is true of the git repository and the source release, but that file is not available in the binary releases:

    $ wget https://bitcoincore.org/bin/bitcoin-core-0.19.1/bitcoin-0.19.1-x86_64-linux-gnu.tar.gz
    $ tar tzf bitcoin-0.19.1-x86_64-linux-gnu.tar.gz | grep -i rpcauth || echo FILE NOT FOUND
    FILE NOT FOUND

IMO, this file should either be included with the binary releases or it should not be mentioned in the help text.  I think this issue needs a decision from experienced contributors about which approach to take, and then can be tagged as a *good first issue* for possible implementation by a less experienced contributor.

Note: this issue is a deliberate duplicate of #14292 (by [request](https://github.com/bitcoin/bitcoin/issues/14292#issuecomment-634733333)).

### PR DESCRIPTION
**Pull request by fanquake**
Fixes #19081.
Addresses: https://github.com/bitcoin/bitcoin/pull/22235#discussion_r946359661.

Guix Build (x86_64):
```bash
bdc14efe02553703f3f27438b17ee452fc0b08e37f626830857c60a7740ec990  guix-build-495caa9a0149/output/aarch64-linux-gnu/SHA256SUMS.part
0c541efd8e2e52f1915cab13c935f165f906e46590175d568be34e026b5f0d9e  guix-build-495caa9a0149/output/aarch64-linux-gnu/bitcoin-495caa9a0149-aarch64-linux-gnu-debug.tar.gz
5ee15e11e0d52d5a2df6b6a60e4ae5b3adc4b259294ad53452478f28efbe70c8  guix-build-495caa9a0149/output/aarch64-linux-gnu/bitcoin-495caa9a0149-aarch64-linux-gnu.tar.gz
451152a282fd79db45a09e0ccb3491e3044ea5ad0931dfe2820bb5c45839a311  guix-build-495caa9a0149/output/arm-linux-gnueabihf/SHA256SUMS.part
b88159ac78d427556e7f70e5da3b849da5491c98fa08c16c25adbfa8e3b44b58  guix-build-495caa9a0149/output/arm-linux-gnueabihf/bitcoin-495caa9a0149-arm-linux-gnueabihf-debug.tar.gz
54b59e4d06144d18b97a0260330383dafdcf3473b560c71f6304a8d1b9ef0100  guix-build-495caa9a0149/output/arm-linux-gnueabihf/bitcoin-495caa9a0149-arm-linux-gnueabihf.tar.gz
784f43a651ed6f9335117ca971c7c215eba536414792edf72a0dc372c901c3fa  guix-build-495caa9a0149/output/arm64-apple-darwin/SHA256SUMS.part
ace525b20dc4789c931cf781027680e5fe35bd987c8d1908590cd83ce7e44495  guix-build-495caa9a0149/output/arm64-apple-darwin/bitcoin-495caa9a0149-arm64-apple-darwin-unsigned.dmg
1ab6d5d91040567e4fccb4e2de49789f78bd1ce9b4a496486d837904591bfbc6  guix-build-495caa9a0149/output/arm64-apple-darwin/bitcoin-495caa9a0149-arm64-apple-darwin-unsigned.tar.gz
a5cd681e67beda05cac83d133d6996699effb12b9830b269669057d02b0a4eef  guix-build-495caa9a0149/output/arm64-apple-darwin/bitcoin-495caa9a0149-arm64-apple-darwin.tar.gz
01cc3a078a952ed9bbff38e79bad4885a5f8502906498f7b909cb2cc798ada64  guix-build-495caa9a0149/output/dist-archive/bitcoin-495caa9a0149.tar.gz
d4652f912573d0c9ce11d7e84fad10ad03f67247f85b0a5d9c47db74508705ce  guix-build-495caa9a0149/output/powerpc64-linux-gnu/SHA256SUMS.part
76facbb81819b2d81afd61e2234b65d99c9e28b28f2b1d3ee2d7119cf192ba60  guix-build-495caa9a0149/output/powerpc64-linux-gnu/bitcoin-495caa9a0149-powerpc64-linux-gnu-debug.tar.gz
a75cef05abb05da4a1ba81669f6a66ecc790946b6b2fd3749bad0b462bb50219  guix-build-495caa9a0149/output/powerpc64-linux-gnu/bitcoin-495caa9a0149-powerpc64-linux-gnu.tar.gz
2b2ac79848a643f025960cabca2694cb3537d72d83477ed1b53bcc6e2ee11ce4  guix-build-495caa9a0149/output/powerpc64le-linux-gnu/SHA256SUMS.part
d0542623109fb72d72dcd06f8de30edefdd7e0f1bf46a7be7c262c0e631c49d8  guix-build-495caa9a0149/output/powerpc64le-linux-gnu/bitcoin-495caa9a0149-powerpc64le-linux-gnu-debug.tar.gz
b85dbcbe4f50cbddcd167568d31326d62e786720cb237aaf3d11b69394f01983  guix-build-495caa9a0149/output/powerpc64le-linux-gnu/bitcoin-495caa9a0149-powerpc64le-linux-gnu.tar.gz
a85a00d353338d2e72a4218782e05a816f21aea0637eb21742c55ee9a9ce1f7b  guix-build-495caa9a0149/output/riscv64-linux-gnu/SHA256SUMS.part
e10561b08a91a22947ead27b6d4a07581fa566eb3e5d8161f97b2ff2708b85f4  guix-build-495caa9a0149/output/riscv64-linux-gnu/bitcoin-495caa9a0149-riscv64-linux-gnu-debug.tar.gz
994aca800c2580349a436fb1fa8195cd943363dee5c3ee7c91d726cc95cd3068  guix-build-495caa9a0149/output/riscv64-linux-gnu/bitcoin-495caa9a0149-riscv64-linux-gnu.tar.gz
5a063e6dbf64aca4a639fea10fcff2f7a533819607558cec25a9a88bc575c0dc  guix-build-495caa9a0149/output/x86_64-apple-darwin/SHA256SUMS.part
e7f36749c3a48ce7173ee100abb275eb217eee2713d52d8b576f8c749e0c2e9e  guix-build-495caa9a0149/output/x86_64-apple-darwin/bitcoin-495caa9a0149-x86_64-apple-darwin-unsigned.dmg
b0289c6caba5e092d8e287c9cb4f328844bce04fce74de91e76821423b7b8153  guix-build-495caa9a0149/output/x86_64-apple-darwin/bitcoin-495caa9a0149-x86_64-apple-darwin-unsigned.tar.gz
2bf983abcfd0d782632ef7fb116ba169a0b282264774c032a50beca016468ed0  guix-build-495caa9a0149/output/x86_64-apple-darwin/bitcoin-495caa9a0149-x86_64-apple-darwin.tar.gz
72705350ab34f8d33ef98e5563f36aa82f15e7b450e3914f591b9c88e8b385da  guix-build-495caa9a0149/output/x86_64-linux-gnu/SHA256SUMS.part
7b5285bfd28acf8ae1471807962d31857afe70e13ae087a8aec83d5a8ac1a184  guix-build-495caa9a0149/output/x86_64-linux-gnu/bitcoin-495caa9a0149-x86_64-linux-gnu-debug.tar.gz
6544591754607792b731fe630bf8235e78e08c7edb4d1d210b35d6328240acc1  guix-build-495caa9a0149/output/x86_64-linux-gnu/bitcoin-495caa9a0149-x86_64-linux-gnu.tar.gz
bbd6490489ddfde06b6cde602d317d84c09495de3cf033016ea0ba5e60770fdb  guix-build-495caa9a0149/output/x86_64-w64-mingw32/SHA256SUMS.part
f48f6efcf5822517a697d990eddbc03011b08c5e4be4886e25ab65c8a0951aaa  guix-build-495caa9a0149/output/x86_64-w64-mingw32/bitcoin-495caa9a0149-win64-debug.zip
b1a83e58d7b316c4743ae16d552f615aa710a4985ded5834d1de6fe6437ca3b2  guix-build-495caa9a0149/output/x86_64-w64-mingw32/bitcoin-495caa9a0149-win64-setup-unsigned.exe
0196abf8cb27f3cb03b4688a85a97c26975db4938f7a2dc0a09f11847f6530bb  guix-build-495caa9a0149/output/x86_64-w64-mingw32/bitcoin-495caa9a0149-win64-unsigned.tar.gz
426994640e85f82b3d3bee25f07930f952edf4a60ecd7ba41f83c82cbbcc63d1  guix-build-495caa9a0149/output/x86_64-w64-mingw32/bitcoin-495caa9a0149-win64.zip
```

Guix Build (arm64):
```bash
506a2fade794398ec38cf9c59ca0b454fdcbb0080d7e40f1ace698a7b08eef77  guix-build-cf39913e5748/output/arm-linux-gnueabihf/SHA256SUMS.part
8ec3686ba80709eb024ea02b7b493e70af82383d814b105f157c94a86cb7efd6  guix-build-cf39913e5748/output/arm-linux-gnueabihf/bitcoin-cf39913e5748-arm-linux-gnueabihf-debug.tar.gz
4d9b3456c974f7f47dca64bcc994fa04888bf08518c902d12366798829a9100d  guix-build-cf39913e5748/output/arm-linux-gnueabihf/bitcoin-cf39913e5748-arm-linux-gnueabihf.tar.gz
7c74610dcaa1dbb7fcadb2ad59520fdfa3fa02f26ceb88ada178ac35a0644840  guix-build-cf39913e5748/output/arm64-apple-darwin/SHA256SUMS.part
fca59c367d2b5891186c546bd87ed66dcc4f0b11433b54c49ad72aa5e0e49978  guix-build-cf39913e5748/output/arm64-apple-darwin/bitcoin-cf39913e5748-arm64-apple-darwin-unsigned.dmg
467e5c2eeaccac995d14b41f5418ba783b31ee7c63d76d5fce5e667bd6ef6683  guix-build-cf39913e5748/output/arm64-apple-darwin/bitcoin-cf39913e5748-arm64-apple-darwin-unsigned.tar.gz
1f19caa059cd6b4adcc089f90ce33fe7f679b3a4beb970b3ef5c3292fe5841ee  guix-build-cf39913e5748/output/arm64-apple-darwin/bitcoin-cf39913e5748-arm64-apple-darwin.tar.gz
ea07945bf21cbe306e8afc815ccb5f4a363fb7335239fe35b3668bdef43f735a  guix-build-cf39913e5748/output/dist-archive/bitcoin-cf39913e5748.tar.gz
b699ae65b899192ad5afbe6a069d2113fda92b833f7db31a1487e7c4c04440f5  guix-build-cf39913e5748/output/powerpc64-linux-gnu/SHA256SUMS.part
eeced64cd3e49ecd95ec29623f7cd3afe66ceffb8296af646ee4e1f3fe658ab0  guix-build-cf39913e5748/output/powerpc64-linux-gnu/bitcoin-cf39913e5748-powerpc64-linux-gnu-debug.tar.gz
20181015752a5d7bfc0d173fefb554b785ca0e566ffef443da6673d6a63ef10d  guix-build-cf39913e5748/output/powerpc64-linux-gnu/bitcoin-cf39913e5748-powerpc64-linux-gnu.tar.gz
59338187f551f1c043e80f63c897899365fca4872b4d793e71423c1aa7ff1358  guix-build-cf39913e5748/output/powerpc64le-linux-gnu/SHA256SUMS.part
33d5d3113f176da881df06f5121d6003ae7f4e2d8124da805db1edab39161e81  guix-build-cf39913e5748/output/powerpc64le-linux-gnu/bitcoin-cf39913e5748-powerpc64le-linux-gnu-debug.tar.gz
b5e9d3bf3e02aa6e3419c9f35118cb71323e0e01280b907e48fc335f8981cb6d  guix-build-cf39913e5748/output/powerpc64le-linux-gnu/bitcoin-cf39913e5748-powerpc64le-linux-gnu.tar.gz
ef5d9f2449ad570221209194af4c4824fc70cfacbaa020da24b9186212b81c5f  guix-build-cf39913e5748/output/riscv64-linux-gnu/SHA256SUMS.part
aafee42808b8bb2cdccbd919470bee2a1c9fcbb05daea32d15aa56288db7fc09  guix-build-cf39913e5748/output/riscv64-linux-gnu/bitcoin-cf39913e5748-riscv64-linux-gnu-debug.tar.gz
72e2f3c9a48f8f3f14ecac2e6a8645253a32bfad8a857d13824aedd1796cad6f  guix-build-cf39913e5748/output/riscv64-linux-gnu/bitcoin-cf39913e5748-riscv64-linux-gnu.tar.gz
8637b4e85eb2d7be803e813fbf15ee597aa35c9910b4b413d657fa69c4d4569d  guix-build-cf39913e5748/output/x86_64-apple-darwin/SHA256SUMS.part
7b384d2734bfd650f3f8096aa32b2aa6bddfee137c3e27ac75736989f7e6740e  guix-build-cf39913e5748/output/x86_64-apple-darwin/bitcoin-cf39913e5748-x86_64-apple-darwin-unsigned.dmg
06960637c9322c3b97bb58331b3282e770b02b7b4d95100866a96c31400a9bb9  guix-build-cf39913e5748/output/x86_64-apple-darwin/bitcoin-cf39913e5748-x86_64-apple-darwin-unsigned.tar.gz
cc524e73d9441fc0e396c6ae5847d9b395d27f8bfe65703520792ea509191e2e  guix-build-cf39913e5748/output/x86_64-apple-darwin/bitcoin-cf39913e5748-x86_64-apple-darwin.tar.gz
d99a7a48aa3a226d7f87c26dd4d6f12109e71fcb9f41052c0bcf5ba4bad1c408  guix-build-cf39913e5748/output/x86_64-linux-gnu/SHA256SUMS.part
f2598fedf273244499f9481253f5114db2e299e132f5378c312f684d0a190f5f  guix-build-cf39913e5748/output/x86_64-linux-gnu/bitcoin-cf39913e5748-x86_64-linux-gnu-debug.tar.gz
2151744dae55c460ad0c919101fcb845872060eef957d3ae0672f12098fa87b0  guix-build-cf39913e5748/output/x86_64-linux-gnu/bitcoin-cf39913e5748-x86_64-linux-gnu.tar.gz
6087d880cced5f30d9b70b592c91efc5677f470f82cc9ed00e0fc112ff36ac43  guix-build-cf39913e5748/output/x86_64-w64-mingw32/SHA256SUMS.part
e91e48edb8819cd25b6d6b5f5421473c56859dbb17cf32108e906877097d454d  guix-build-cf39913e5748/output/x86_64-w64-mingw32/bitcoin-cf39913e5748-win64-debug.zip
ee4987cb94dcf4b5dfee65341a7a4f7b0865f1466c715c033a05591a749d43c3  guix-build-cf39913e5748/output/x86_64-w64-mingw32/bitcoin-cf39913e5748-win64-setup-unsigned.exe
43001b416664ea227d3f07100d8a60bcd20c761770a332374a1466d7af734127  guix-build-cf39913e5748/output/x86_64-w64-mingw32/bitcoin-cf39913e5748-win64-unsigned.tar.gz
bb9a56856800e7d6e5c602302ec90704cb5dd917f52d122d783deee3ab1f149d  guix-build-cf39913e5748/output/x86_64-w64-mingw32/bitcoin-cf39913e5748-win64.zip
```

### CODE REVIEW

**Comment by jarolrod** [comment]
GUIX hashes

x86:
```

```

arm64:
```
$ env HOSTS='arm-linux-gnueabihf arm64-apple-darwin powerpc64-linux-gnu powerpc64le-linux-gnu riscv64-linux-gnu x86_64-apple-darwin x86_64-linux-gnu x86_64-w64-mingw32' ./contrib/guix/guix-build
$ find guix-build-$(git rev-parse --short=12 HEAD)/output/ -type f -print0 | env LC_ALL=C sort -z | xargs -r0 sha256sum

96e74f1a8139d82efb5e534640159b9e24c84caee244a53d17c647e796f3cb03  guix-build-71d6765405af/output/arm-linux-gnueabihf/SHA256SUMS.part
65a1c84ec9ffbbf4cb19c0c5f458d3569fb0264066e40457eab5927817f9cb5d  guix-build-71d6765405af/output/arm-linux-gnueabihf/bitcoin-71d6765405af-arm-linux-gnueabihf-debug.tar.gz
8c5cf767e64c5484ab84cf93953f13b5c57a1fe85b8177eb4b146d90be1cdfc3  guix-build-71d6765405af/output/arm-linux-gnueabihf/bitcoin-71d6765405af-arm-linux-gnueabihf.tar.gz
3fb27a95e98ae63cd04eff2622baadb15680a83cbbe4aa53b0b7e25c2980fb07  guix-build-71d6765405af/output/arm64-apple-darwin/SHA256SUMS.part
13958d0829d6fd91e38fdba5c410f485a60f94cf8797711a92b4015601ca79aa  guix-build-71d6765405af/output/arm64-apple-darwin/bitcoin-71d6765405af-arm64-apple-darwin-unsigned.dmg
cdb8b8c3a21cc208ef20f7f65a64eb2c391bef6ff08aeffdbfef453c36dab9bc  guix-build-71d6765405af/output/arm64-apple-darwin/bitcoin-71d6765405af-arm64-apple-darwin-unsigned.tar.gz
94b47f81ff93b571a36506403b08df395f6d34527cd795029be4addc492c4e85  guix-build-71d6765405af/output/arm64-apple-darwin/bitcoin-71d6765405af-arm64-apple-darwin.tar.gz
ad388467efddf908800c7ffa3ab583ebc75fcd59b50e35a15620b60caf5bc60c  guix-build-71d6765405af/output/dist-archive/bitcoin-71d6765405af.tar.gz
14be9157621435d48103ea73f1df8e2b9cda1a3925eeab748edb607897a6a1db  guix-build-71d6765405af/output/powerpc64-linux-gnu/SHA256SUMS.part
dedf03b456e5daaa1658b6f28f600612b82c6d63651e5db8d75901afa23218de  guix-build-71d6765405af/output/powerpc64-linux-gnu/bitcoin-71d6765405af-powerpc64-linux-gnu-debug.tar.gz
c51997a0a1fef6c6a2523dff8eb2ce99e2ebf96de47b02b4ac9dceef905acb90  guix-build-71d6765405af/output/powerpc64-linux-gnu/bitcoin-71d6765405af-powerpc64-linux-gnu.tar.gz
9789b464c7791009fe911437bfeea37ab845982bfa7e0f881322e0135bb8900e  guix-build-71d6765405af/output/powerpc64le-linux-gnu/SHA256SUMS.part
c2039c2ea6cda15e681abfba23a111cba313f7254f776588f8aa5ceef611f379  guix-build-71d6765405af/output/powerpc64le-linux-gnu/bitcoin-71d6765405af-powerpc64le-linux-gnu-debug.tar.gz
8efb09da2d72f4305620f98729fa1bc940ae9ec848d54d55ec856658544683b4  guix-build-71d6765405af/output/powerpc64le-linux-gnu/bitcoin-71d6765405af-powerpc64le-linux-gnu.tar.gz
07a964418dcf79c39e8c71abddcfa27488ee4fbda6900185fda7b4b2977ff58e  guix-build-71d6765405af/output/riscv64-linux-gnu/SHA256SUMS.part
290f7a321d1a1475462f89b506969f153c39f11af128f4369fb15157f59fb951  guix-build-71d6765405af/output/riscv64-linux-gnu/bitcoin-71d6765405af-riscv64-linux-gnu-debug.tar.gz
ac67bbe08e76a7704341c0cc7a030792f4fd4bf1773b396b7ebeb5d1a34bf87e  guix-build-71d6765405af/output/riscv64-linux-gnu/bitcoin-71d6765405af-riscv64-linux-gnu.tar.gz
469ccc48acf56bb9fac9d0a3d8f949e88126dc48d4a88d2066fa5282d253bc6f  guix-build-71d6765405af/output/x86_64-apple-darwin/SHA256SUMS.part
5e587a77e44c4c86f6f2066581f6f757fa56289888b88c414ae148ec67b42e41  guix-build-71d6765405af/output/x86_64-apple-darwin/bitcoin-71d6765405af-x86_64-apple-darwin-unsigned.dmg
e61bc7e26e500496333cd3b3a42712a7fd72a590e5fe1d49787cfdb255231131  guix-build-71d6765405af/output/x86_64-apple-darwin/bitcoin-71d6765405af-x86_64-apple-darwin-unsigned.tar.gz
8ccb31d80f0c2ad91241a89ebb2c62cc4881cfa944b882782aaaadc8bac3385b  guix-build-71d6765405af/output/x86_64-apple-darwin/bitcoin-71d6765405af-x86_64-apple-darwin.tar.gz
03eceb245cfdd28da6986b098e3fa545d35852713f7605035ccb3052f28d4a7e  guix-build-71d6765405af/output/x86_64-linux-gnu/SHA256SUMS.part
e2d9e64656d5fe4c5aeb7bd458a5913b1a76f6453f7acf101be0a6331fd90482  guix-build-71d6765405af/output/x86_64-linux-gnu/bitcoin-71d6765405af-x86_64-linux-gnu-debug.tar.gz
ded765b3854611b91d66eaa84b235ff2f83669bae3b02897a8f2749c2d7ae761  guix-build-71d6765405af/output/x86_64-linux-gnu/bitcoin-71d6765405af-x86_64-linux-gnu.tar.gz
8d5ef626affbc1f275a21323f24fb3b7ee480e7dacef8695fe45687027dbfadf  guix-build-71d6765405af/output/x86_64-w64-mingw32/SHA256SUMS.part
5739340074c2906c06193445a1d1058a8dadd80a245dfa54f7286346944a7a06  guix-build-71d6765405af/output/x86_64-w64-mingw32/bitcoin-71d6765405af-win64-debug.zip
5dcb6ad8adb573c74455bb6ee6572a11e11858bc93ab259c1e08e048c7093076  guix-build-71d6765405af/output/x86_64-w64-mingw32/bitcoin-71d6765405af-win64-setup-unsigned.exe
ea368b8af5e0b034f91582971fa92823ee498a99c48bb74a3e2ca99478bb4ec2  guix-build-71d6765405af/output/x86_64-w64-mingw32/bitcoin-71d6765405af-win64-unsigned.tar.gz
fbb8990baf2f1bd0c79e4534263527d58d2f3879d68f8bd47718a44d0fb76e9e  guix-build-71d6765405af/output/x86_64-w64-mingw32/bitcoin-71d6765405af-win64.zip
```

File: contrib/guix/libexec/build.sh [hunk 1]
```diff
@@ -369,6 +369,8 @@ mkdir -p "$DISTSRC"
         # has not been run before buildling, this file will be a stub
         cp "${DISTSRC}/share/examples/bitcoin.conf" "${DISTNAME}/"
 
+        cp -r "${DISTSRC}/share/rpcauth" "${DISTNAME}/share/"
```
Comment (hebasto) [inline contrib/guix/libexec/build.sh:372]: Install a `share\rpcauth` folder using Windows installer as well?

**Review by hebasto — COMMENTED** [review]
Concept ACK.

File: contrib/guix/libexec/build.sh (hunk 1 above)
Comment (fanquake) [reply contrib/guix/libexec/build.sh:372]: Added a (untested) change for this.

File: share/setup.nsi.in [hunk 2]
```diff
@@ -77,6 +77,8 @@ Section -Main SEC0000
     File /oname=readme.txt @abs_top_srcdir@/doc/README_windows.txt
     SetOutPath $INSTDIR\share\rpcauth
     File /r /x @abs_top_srcdir@/share\rpcauth\*.*
+    SetOutPath $INSTDIR\share\examples
+    File @abs_top_srcdir@/share\examples\bitcoin.conf
```
Comment (josibake) [inline share/setup.nsi.in:81]: should this also be included in the uninstall section below? L131-L136

Comment (fanquake) [reply share/setup.nsi.in:81]: It should be covered by removing the share directory?

**Comment by josibake** [comment]
Concept ACK

File: share/setup.nsi.in (hunk 2 above)
Comment (josibake) [reply share/setup.nsi.in:81]: :doh: you're right, i missed that

**Comment by hebasto** [comment]
This diff seem required:
```diff
--- a/share/setup.nsi.in
+++ b/share/setup.nsi.in
@@ -76,9 +76,10 @@ Section -Main SEC0000
     File /oname=COPYING.txt @abs_top_srcdir@/COPYING
     File /oname=readme.txt @abs_top_srcdir@/doc/README_windows.txt
     SetOutPath $INSTDIR\share\rpcauth
-    File /r /x @abs_top_srcdir@/share\rpcauth\*.*
+    File @abs_top_srcdir@/share/rpcauth/rpcauth.py
+    File @abs_top_srcdir@/share/rpcauth/README.md
     SetOutPath $INSTDIR\share\examples
-    File @abs_top_srcdir@/share\examples\bitcoin.conf
+    File @abs_top_srcdir@/share/examples/bitcoin.conf
     SetOutPath $INSTDIR\daemon
     File @abs_top_builddir@/release/@BITCOIN_DAEMON_NAME@@EXEEXT@
     File @abs_top_builddir@/release/@BITCOIN_CLI_NAME@@EXEEXT@
```

**Comment by fanquake** [comment]
> This diff seem required:

Incorporated this, and moved bitcoin.conf to the root of the install directory.

**Comment by hebasto** [comment]
I've re-read [docs](https://nsis.sourceforge.io/Docs/Chapter4.html#file), only `/x` was wrong:
```diff
--- a/share/setup.nsi.in
+++ b/share/setup.nsi.in
@@ -76,7 +76,7 @@ Section -Main SEC0000
     File /oname=COPYING.txt @abs_top_srcdir@/COPYING
     File /oname=readme.txt @abs_top_srcdir@/doc/README_windows.txt
     SetOutPath $INSTDIR\share\rpcauth
-    File /r /x @abs_top_srcdir@/share\rpcauth\*.*
+    File /r @abs_top_srcdir@/share\rpcauth
     SetOutPath $INSTDIR\share\examples
     File @abs_top_srcdir@/share\examples\bitcoin.conf
     SetOutPath $INSTDIR\daemon
```

**Comment by hebasto** [comment]
Guix build on `x86_64`:
```
b5b4b56ec4dca620c099846e583cfb5c5c6421e17009bddc49edb678136e92eb  guix-build-3741c1c7fb5f/output/dist-archive/bitcoin-3741c1c7fb5f.tar.gz
565dba01ac1426acd7b20d9937c4ced1c3181de530e96a398afd624dc199a6f0  guix-build-3741c1c7fb5f/output/x86_64-w64-mingw32/SHA256SUMS.part
28fd51e1a39af3260e39e984e0f02e27f772db28e728df4a8cc20d9f74cd90d4  guix-build-3741c1c7fb5f/output/x86_64-w64-mingw32/bitcoin-3741c1c7fb5f-win64-debug.zip
55d7799584daf317011f3b8bfd0ec11769f75d0caa4b7736fe59be25b6b7861e  guix-build-3741c1c7fb5f/output/x86_64-w64-mingw32/bitcoin-3741c1c7fb5f-win64-setup-unsigned.exe
88186964d0d71d53e3d5b22f4d1ecfc54fc992aae46b6484d156071a6960afe4  guix-build-3741c1c7fb5f/output/x86_64-w64-mingw32/bitcoin-3741c1c7fb5f-win64-unsigned.tar.gz
c668b5b8a0439a6b5c81c0ac994c3fc8424d5e74ac152fe78c372793ea6d30e4  guix-build-3741c1c7fb5f/output/x86_64-w64-mingw32/bitcoin-3741c1c7fb5f-win64.zip
```

**Review by hebasto — COMMENTED** [review]
Tested `bitcoin-3741c1c7fb5f-win64-setup-unsigned.exe` tested on Windows 11:

```
C:\Program Files\Bitcoin>tree /f
Folder PATH listing
Volume serial number is 2850-CFA9
C:.
│   bitcoin-qt.exe
│   bitcoin.conf
│   COPYING.txt
│   readme.txt
│   uninstall.exe
│
├───daemon
│       bitcoin-cli.exe
│       bitcoin-tx.exe
│       bitcoin-wallet.exe
│       bitcoind.exe
│       test_bitcoin.exe
│
└───share
    └───rpcauth
        └───rpcauth
                README.md
                rpcauth.py

```

**Comment by hebasto** [comment]
The following patch fixes `share\rpcauth\rpcauth` path:
```diff
--- a/share/setup.nsi.in
+++ b/share/setup.nsi.in
@@ -77,7 +77,7 @@ Section -Main SEC0000
     File /oname=readme.txt @abs_top_srcdir@/doc/README_windows.txt
     File @abs_top_srcdir@/share/examples/bitcoin.conf
     SetOutPath $INSTDIR\share\rpcauth
-    File /r @abs_top_srcdir@/share\rpcauth
+    File @abs_top_srcdir@/share/rpcauth/*.*
     SetOutPath $INSTDIR\daemon
     File @abs_top_builddir@/release/@BITCOIN_DAEMON_NAME@@EXEEXT@
     File @abs_top_builddir@/release/@BITCOIN_CLI_NAME@@EXEEXT@
```

I'm really sorry about the mess I created in this PR.

**Review by hebasto — APPROVED** [review]
ACK 495caa9a0149a87c4148fbae1b622c998c019d40, tested `bitcoin-495caa9a0149-win64-setup-unsigned.exe` on Windows 11 Pro:

```
C:\Program Files\Bitcoin>tree /f
Folder PATH listing
Volume serial number is 2850-CFA9
C:.
│   bitcoin-qt.exe
│   bitcoin.conf
│   COPYING.txt
│   readme.txt
│   uninstall.exe
│
├───daemon
│       bitcoin-cli.exe
│       bitcoin-tx.exe
│       bitcoin-wallet.exe
│       bitcoind.exe
│       test_bitcoin.exe
│
└───share
    └───rpcauth
            README.md
            rpcauth.py


```

Also an uninstaller tested.

Guix build on `x86_64`:
```
01cc3a078a952ed9bbff38e79bad4885a5f8502906498f7b909cb2cc798ada64  guix-build-495caa9a0149/output/dist-archive/bitcoin-495caa9a0149.tar.gz
bbd6490489ddfde06b6cde602d317d84c09495de3cf033016ea0ba5e60770fdb  guix-build-495caa9a0149/output/x86_64-w64-mingw32/SHA256SUMS.part
f48f6efcf5822517a697d990eddbc03011b08c5e4be4886e25ab65c8a0951aaa  guix-build-495caa9a0149/output/x86_64-w64-mingw32/bitcoin-495caa9a0149-win64-debug.zip
b1a83e58d7b316c4743ae16d552f615aa710a4985ded5834d1de6fe6437ca3b2  guix-build-495caa9a0149/output/x86_64-w64-mingw32/bitcoin-495caa9a0149-win64-setup-unsigned.exe
0196abf8cb27f3cb03b4688a85a97c26975db4938f7a2dc0a09f11847f6530bb  guix-build-495caa9a0149/output/x86_64-w64-mingw32/bitcoin-495caa9a0149-win64-unsigned.tar.gz
426994640e85f82b3d3bee25f07930f952edf4a60ecd7ba41f83c82cbbcc63d1  guix-build-495caa9a0149/output/x86_64-w64-mingw32/bitcoin-495caa9a0149-win64.zip
```

**Review by josibake — APPROVED** [review]
ACK https://github.com/bitcoin/bitcoin/pull/25829/commits/495caa9a0149a87c4148fbae1b622c998c019d40

Don't have access to a windows machine to test the exe but here is the guix build:

```
01cc3a078a952ed9bbff38e79bad4885a5f8502906498f7b909cb2cc798ada64  guix-build-495caa9a0149/output/dist-archive/bitcoin-495caa9a0149.tar.gz
bbd6490489ddfde06b6cde602d317d84c09495de3cf033016ea0ba5e60770fdb  guix-build-495caa9a0149/output/x86_64-w64-mingw32/SHA256SUMS.part
f48f6efcf5822517a697d990eddbc03011b08c5e4be4886e25ab65c8a0951aaa  guix-build-495caa9a0149/output/x86_64-w64-mingw32/bitcoin-495caa9a0149-win64-debug.zip
b1a83e58d7b316c4743ae16d552f615aa710a4985ded5834d1de6fe6437ca3b2  guix-build-495caa9a0149/output/x86_64-w64-mingw32/bitcoin-495caa9a0149-win64-setup-unsigned.exe
0196abf8cb27f3cb03b4688a85a97c26975db4938f7a2dc0a09f11847f6530bb  guix-build-495caa9a0149/output/x86_64-w64-mingw32/bitcoin-495caa9a0149-win64-unsigned.tar.gz
426994640e85f82b3d3bee25f07930f952edf4a60ecd7ba41f83c82cbbcc63d1  guix-build-495caa9a0149/output/x86_64-w64-mingw32/bitcoin-495caa9a0149-win64.zip
```

**Comment by jarolrod** [comment]
Post merge tACK

guix hashes: 
```
$ find guix-build-$(git rev-parse --short=12 HEAD)/output/ -type f -print0 | env LC_ALL=C sort -z | xargs -r0 sha256sum

01cc3a078a952ed9bbff38e79bad4885a5f8502906498f7b909cb2cc798ada64  guix-build-495caa9a0149/output/dist-archive/bitcoin-495caa9a0149.tar.gz
bbd6490489ddfde06b6cde602d317d84c09495de3cf033016ea0ba5e60770fdb  guix-build-495caa9a0149/output/x86_64-w64-mingw32/SHA256SUMS.part
f48f6efcf5822517a697d990eddbc03011b08c5e4be4886e25ab65c8a0951aaa  guix-build-495caa9a0149/output/x86_64-w64-mingw32/bitcoin-495caa9a0149-win64-debug.zip
b1a83e58d7b316c4743ae16d552f615aa710a4985ded5834d1de6fe6437ca3b2  guix-build-495caa9a0149/output/x86_64-w64-mingw32/bitcoin-495caa9a0149-win64-setup-unsigned.exe
0196abf8cb27f3cb03b4688a85a97c26975db4938f7a2dc0a09f11847f6530bb  guix-build-495caa9a0149/output/x86_64-w64-mingw32/bitcoin-495caa9a0149-win64-unsigned.tar.gz
426994640e85f82b3d3bee25f07930f952edf4a60ecd7ba41f83c82cbbcc63d1  guix-build-495caa9a0149/output/x86_64-w64-mingw32/bitcoin-495caa9a0149-win64.zip
```

**Pull request merged**